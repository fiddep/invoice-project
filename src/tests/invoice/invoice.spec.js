const createInvoice = require('../../domain/service/invoice/createInvoice');

const CustomerRepository = require('../../domain/repository/CustomerRepository');
const OrderRepository = require('../../domain/repository/OrderRepository');
const UserRepository = require('../../domain/repository/UserRepository');
const InvoiceRepository = require('../../domain/repository/InvoiceRepository');

const mockOrderRepository = new OrderRepository();
const mockUserRepository = new UserRepository();
const mockCustomerRepository = new CustomerRepository();
const mockInvoiceRepository = new InvoiceRepository();

describe('createInvoice', () => {
  it('create invoice from all parts', async () => {
    Date.now = jest.fn(() => 1482363367071);
    const mockId = 0;
    mockOrderRepository.get = jest.fn(id => ({}));
    mockUserRepository.get = jest.fn(id => ({
      settings: { invoiceStrategy: 'linear-increment' },
    }));
    mockCustomerRepository.get = jest.fn(id => ({}));
    mockInvoiceRepository.persist = jest.fn(obj => obj);
    mockInvoiceRepository.last = jest.fn(() => ({ id: 2 }));

    const invoice = await createInvoice(mockId, mockId, mockId, {
      orderRepository: mockOrderRepository,
      userRepository: mockUserRepository,
      customerRepository: mockCustomerRepository,
      invoiceRepository: mockInvoiceRepository,
    });

    expect(mockInvoiceRepository.persist).toHaveBeenCalled();
    expect(invoice).toMatchSnapshot();
  });
});
