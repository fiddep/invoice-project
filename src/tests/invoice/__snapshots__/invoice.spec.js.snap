// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createInvoice create invoice from all parts 1`] = `
Invoice {
  "createdAt": 1482363367071,
  "customer": Object {},
  "id": null,
  "invoiceId": "3",
  "order": Object {},
  "paymentTerm": 20,
  "user": Object {
    "settings": Object {
      "invoiceStrategy": "linear-increment",
    },
  },
}
`;
